spock-core/src/main/java/org/spockframework/mock/runtime/JavaMockFactory.java

```diff
@@ -45,25 +47,34 @@ public Object createDetached(IMockConfiguration configuration, ClassLoader class
 	}
 
   private Object createInternal(IMockConfiguration configuration, Specification specification, ClassLoader classLoader) {
-    if (Modifier.isFinal(configuration.getType().getModifiers())) {
-      throw new CannotCreateMockException(configuration.getType(),
+    final Class<?> type = configuration.getType();
+    if (Modifier.isFinal(type.getModifiers())) {
+      throw new CannotCreateMockException(type,
           " because Java mocks cannot mock final classes. If the code under test is written in Groovy, use a Groovy mock.");
     }
     if (configuration.isGlobal()) {
-      throw new CannotCreateMockException(configuration.getType(),
+      throw new CannotCreateMockException(type,
           " because Java mocks cannot mock globally. If the code under test is written in Groovy, use a Groovy mock.");
     }
 
-    MetaClass mockMetaClass = GroovyRuntimeUtil.getMetaClass(configuration.getType());
-    IProxyBasedMockInterceptor interceptor = new JavaMockInterceptor(configuration, specification, mockMetaClass);
-    Object proxy = ProxyBasedMockFactory.INSTANCE.create(configuration.getType(), configuration.getAdditionalInterfaces(),
+    MetaClass mockMetaClass = GroovyRuntimeUtil.getMetaClass(type);
+    JavaMockInterceptor interceptor = new JavaMockInterceptor(configuration, specification, mockMetaClass);
+    List<Class<?>> additionalInterfaces = configuration.getAdditionalInterfaces();
+    Object proxy = ProxyBasedMockFactory.INSTANCE.create(type, additionalInterfaces,
       configuration.getConstructorArgs(), interceptor, classLoader,
       configuration.isUseObjenesis());
+
+    if (!additionalInterfaces.isEmpty() && GroovyObject.class.isAssignableFrom(type)) {
+      //Ticket #1405: We need to update the mockMetaClass to reflect the methods of the additional interfaces
+      //              The MetaClass of the mock is a bit too much, but we do not have a class representing the hierarchy without the internal Spock interfaces like ISpockMockObject
```

Comment:
Yes we could do that, but the question is, if a user would really see the difference, because when a User asks e.g. reflection he would always see the internal Spock stuff. So for me that does not make such a big difference, if the meta-class also surfaces that.
Also if a user would ask the mock for `mock instanceof ISpockMockObject` it would also answer with `true` regardless of, if we add another abstract class/ interface layer there.

So I would say it is not worth the effort.